src/zh/sixmh/src/eu/kanade/tachiyomi/extension/zh/sixmh/SixMH.kt

```diff
@@ -1,38 +1,87 @@
 package eu.kanade.tachiyomi.extension.zh.sixmh
 
-import android.app.Application
-import android.os.Build
-import eu.kanade.tachiyomi.multisrc.mccms.MCCMS
+import android.net.Uri
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
 import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
+import okhttp3.Request
 import okhttp3.Response
-import uy.kohesive.injekt.Injekt
-import uy.kohesive.injekt.api.get
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
 import uy.kohesive.injekt.injectLazy
 
-class SixMH : MCCMS("六漫画", "https://www.liumanhua.com") {
-    private val dataRegex = Regex("var DATA = '([A-Za-z0-9+/=]+)'")
+class SixMH : SimpleParsedHttpSource() {
+    private val paramsRegex = Regex("params = '([A-Za-z0-9+/=]+)'")
     private val json by injectLazy<Json>()
-    override val versionId get() = 2
+    override val versionId get() = 3
+    override val name: String = "六漫画"
+    override val lang: String = "zh"
+    override val supportsLatest: Boolean = true
+    override val baseUrl: String = "https://www.liumanhua.com"
 
-    init {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            // Delete old preferences for "6漫画/zh/1"
-            Injekt.get<Application>().deleteSharedPreferences("source_7259486566651312186")
-        }
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/category/order/hits/page/$page", headers)
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/category/order/addtime/page/$page", headers)
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val uri = Uri.parse(baseUrl).buildUpon()
+            .appendPath("index.php")
+            .appendPath("search")
+            .appendQueryParameter("key", query).toString()
+        return GET(uri, headers)
+    }
+
+    override fun simpleNextPageSelector(): String? = null
+    override fun simpleMangaSelector(): String = "div.cy_list_mh ul"
+    override fun simpleMangaFromElement(element: Element): SManga = SManga.create().apply {
+        title = element.select("li.title > a").text()
+        url = element.select("li.title > a").attr("href")
+        thumbnail_url = element.select("img").attr("src")
```

Comment:
Use `.selectFirst`. Do not add non-null assert for non-critical fields (only title and URL is required).


Comment applies throughout this file.